utils/hardware_info.go

```diff
@@ -0,0 +1,61 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"go.viam.com/utils"
+)
+
+// GetArchitectureInfo returns the architecture of the board.
+func GetArchitectureInfo(modelName string) (utils.StringSet, error) {
```

Comment:
Not sure that the comment matches the code. I think this function returns a set of strings describing compatible board descriptions.